File: Data Structures & Algorithms/house-robber/submission-3.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int sum = 0;
        if(nums.size() == 2){
            return max(nums[0],nums[1]);
        }
        if(nums.size() == 3){
            return max(nums[0],nums[1],nums[2]);
        }
        for(int i=0;i<nums.size();i+=2){
            sum = sum + nums[i];
        }
        return sum;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashMap<Character,Integer>mp = new HashMap<>();
        int left = 0;
        int ans = 0;
        for(int right=0;right<s.length();right++){
            if(mp.containsKey(s.charAt(right))){
                left = Math.max(left , mp.get(s.charAt(right))+1);
            }
            mp.put(s.charAt(right) , right);
            ans = Math.max(ans,right-left+1);
        }
        return ans;
    }
}
